packages/neuron-ui/src/components/Transaction/index.tsx

```diff
@@ -126,6 +137,50 @@ const Transaction = () => {
     )})`
   }, [transaction.outputs.length, transaction.outputsCount, t])
 
+  const onTagClicked = (lockScript: CKBComponents.Script, address: string) => {
+    setShowLockInfoDialog(true)
+    setLockInfo(lockScript)
+    setNewAddress(address)
+  }
+
+  const renderTag = (address: string) => {
+    const lockScript = addressToScript(address)
+    const newAdd = lockScriptToAddress(lockScript, isMainnet)
+    const foundLock = [
+      MultiSigLockInfo,
+      DefaultLockInfo,
+      AnyoneCanPayLockInfoOnAggron,
+      AnyoneCanPayLockInfoOnLina,
+      ChequeLockInfoOnAggron,
+      ChequeLockInfoOnLina,
+    ].find(
+      info =>
+        lockScript.codeHash === info.CodeHash &&
+        lockScript.hashType === info.HashType &&
+        info.ArgsLen.split(',').includes(`${(lockScript.args.length - 2) / 2}`)
+    )
+
+    if (!foundLock) {
+      return null
+    }
+    return (
+      <div className={styles.tagWrap}>
+        <Tag text={foundLock?.TagName} onClick={() => onTagClicked(lockScript, newAdd)} />
```

Comment:
optional chaining in `foundLock?.TagName` is redundant since `!foundLock` short circuit this.

Reply:
Well, optional chaining in `foundLock?.TagName` is what I did at first. Later I added `!foundLock`, but forgot to delete the `?`.